backend/src/tools/web_pages.py

```diff
@@ -1,7 +1,35 @@
 import langchain_core.tools as lcct
+import bs4
+import requests as req
+import re
 
+# Should this perhaps be a separate, deterministic skill that we can plug into 
+#   any tool call that needs to clean text?
+def get_embedded_text(input: bs4.BeautifulSoup, strip:bool = False, separator: str = "") -> str:
```

Comment:
is this used here? if not it shouldn't be here. We will likely implement some RAG in the near future, it could be a better place for it.